pandas/tests/arrays/sparse/test_array.py

```diff
@@ -858,7 +859,21 @@ def test_nonzero(self):
         sa = SparseArray([0, 0, 1, 0, 0, 2, 0, 0, 0, 3, 0, 0])
         (result,) = sa.nonzero()
         tm.assert_numpy_array_equal(expected, result)
-
+    
+    def test_loc(self):
+        # Tests .loc on sparse DataFrame #34687
+        df = pd.DataFrame.sparse.from_spmatrix(eye(5))
+        res1 = df.loc[range(2)].to_numpy()
+        exp1 = pd.DataFrame(
```

Comment:
pls don't use `to_numpy` here. Compare the dataframes using `tm.assert_frame_equal`